staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go

```diff
@@ -46,7 +46,7 @@ import (
 func createHandler(r rest.NamedCreater, scope *RequestScope, admit admission.Interface, includeName bool) http.HandlerFunc {
 	return func(w http.ResponseWriter, req *http.Request) {
 		// For performance tracking purposes.
-		trace := utiltrace.New("Create", utiltrace.Field{"url", req.URL.Path})
+		trace := utiltrace.New("Create", utiltrace.Field{Key: "url", Value: req.URL.Path}, utiltrace.Field{Key: "user-agent", Value: req.UserAgent()}, utiltrace.Field{Key: "remote", Value: req.RemoteAddr})
```

Comment:
how large can user agent be? do we care if it is very large?

When populating for audit events, we use `util/net#SourceIPs`, which takes into account X-Forwarded-For headers, etc. To avoid calling SourceIPs unnecessarily (if it's too expensive), we could wrap with a type that implements `String() string` and lazily calls SourceIPs if needed for output

Reply:
1. In audit events, `user-agent` is truncated in `apiserver/pkg/audit#maybeTruncateUserAgent` if it is very large, how about make this function public in `util/net` and reuse this function?

2. make sense.